### PR TITLE
[Star tree] Rename count to value_count to be inline with aggregations

### PR DESCRIPTION
**Pull request by bharath-techie**
### Description
Rename "count" to "value_count" to be inline with naming conventions of search aggregations

### Related Issues
Solves part of https://github.com/opensearch-project/OpenSearch/issues/15148

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/test/java/org/opensearch/common/util/ByteArrayBackedBitsetTests.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (sarthakaggarwal97) [inline server/src/test/java/org/opensearch/common/util/ByteArrayBackedBitsetTests.java:1]: coming from here: https://github.com/opensearch-project/OpenSearch/pull/14817#discussion_r1704946374

**Comment by sarthakaggarwal97** [comment]
LGTM! thanks @bharath-techie 

Adding here for reference: https://opensearch.org/docs/latest/aggregations/metric/value-count/

**Review by sarthakaggarwal97 — APPROVED** [review]

File: server/src/test/java/org/opensearch/common/util/ByteArrayBackedBitsetTests.java (hunk 1 above)
Comment (bharath-techie) [reply server/src/test/java/org/opensearch/common/util/ByteArrayBackedBitsetTests.java:1]: Yes thanks for linking

**Review by ajaymovva — APPROVED** [review]
LGTM

**Comment by dblock** [comment]
Just confirming that this is not a breaking change for something that we GA-ed already?

**Comment by sarthakaggarwal97** [comment]
> Just confirming that this is not a breaking change for something that we GA-ed already?

@dblock Star Tree is yet to be GA-ed as a feature. It still under development, and the plan is to backport it to 2.x for OS 2.17. We introduced a count as a aggregation [here](https://github.com/opensearch-project/OpenSearch/pull/14605/files#diff-68b3def2747515fd74a8da5df2f94819d9ae508de25692caa6b0856724597422R20), but its still a part of main branch only. 
This PR brings star tree count aggregation inline with present aggregations.

**Comment by dblock** [comment]
@bharath-techie You might want to start drafting API changes in https://github.com/opensearch-project/opensearch-api-specification earlier than later for this feature. That repo supports testing with feature flags enabled, too.

**Comment by bharath-techie** [comment]
> @bharath-techie You might want to start drafting API changes in https://github.com/opensearch-project/opensearch-api-specification earlier than later for this feature. That repo supports testing with feature flags enabled, too.

Sure thanks @dblock let me track this - will create an associated issue as well.

**Review by sachinpkale — APPROVED** [review]

**Review by sachinpkale — APPROVED** [review]

**Pull request merged**